src/Controller/EnMarche/MembershipController.php

```diff
@@ -222,11 +237,77 @@ public function pinInterestsAction(Request $request): Response
                 $this->getDoctrine()->getManager()->flush();
             }
 
-            return $this->redirectToRoute('app_adherent_home');
+            return $this->redirectToRoute('app_membership_choose_committees_around_adherent');
         }
 
         return $this->render('membership/pin_interests.html.twig', [
             'form' => $form->createView(),
         ]);
     }
+
+    /**
+     * This action enables a user to follow some committees during the registration process.
+     *
+     * @Route("/inscription/choisir-des-comites", name="app_membership_choose_committees_around_adherent")
+     * @Method("GET|POST")
+     * @Security("is_granted('MEMBERSHIP_REGISTRATION_IN_PROGRESS')")
+     */
+    public function chooseCommitteesAction(
+        Request $request,
+        AdherentRepository $adherentRepository,
+        CommitteeRepository $committeeRepository,
+        MembershipRegistrationProcess $membershipRegistrationProcess,
+        CoordinatesFactory $coordinatesFactory,
+        CommitteeManager $committeeManager
+    ): Response {
+        /** @var Adherent $adherent */
+        if (!$adherent = $adherentRepository->findByUuid($membershipRegistrationProcess->getAdherentUuid())) {
+            throw $this->createNotFoundException('New adherent not found.');
+        }
+
+        $coordinates = $coordinatesFactory->createFromPostAddress($adherent->getPostAddress());
```

Comment:
why we cannot use `createFromAdherent` method which retrieves the coordinates of the adherent post address?